src/smvlang/parser.y

```diff
@@ -486,17 +486,15 @@ assignments: assignment
 
 assignment : assignment_head '(' assignment_var ')' BECOMES_Token formula ';'
            {
-             binary($$, $3, ID_equal, $6);
-
              if(stack_expr($1).id()==ID_smv_next)
              {
-               exprt &op=to_binary_expr(stack_expr($$)).op0();
+               exprt &op=stack_expr($3);
                unary_exprt tmp(ID_smv_next, std::move(op));
                tmp.swap(op);
-               PARSER.module->add_assign_next(to_equal_expr(stack_expr($$)));
+               PARSER.module->add_assign_next(std::move(stack_expr($3)), std::move(stack_expr($6)));
```

Comment:
How about:
```
unary_exprt tmp{ID_smv_next, stack_expr($3)};
PARSER.module->add_assign_next(std::move(tmp), std::move(stack_expr($6)));
```
even if just a human not having to track what really lives at `$3` at which point in this sequence.